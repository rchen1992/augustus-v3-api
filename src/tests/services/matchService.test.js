const createMatchService = require('@services/matchService');
const mockMatchRepo = require('@tests/mockRepos/match');
const mockLadderUserRepo = require('@tests/mockRepos/ladderUser');

describe('Match Service', () => {
    test('should be able to get user match stats', async () => {
        const service = createMatchService(mockMatchRepo);
        const stats = await service.getLadderUserMatchStats('1', 1);
        expect(stats).toMatchSnapshot();
    });

    test('should be able to get create match', async () => {
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        const match = await service.newMatch(1, '1', '2', '1', '2');
        expect(match).toMatchSnapshot();
    });

    test('should be able to get create match with a tie', async () => {
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        const match = await service.newMatch(1, '1', '2', null, null);
        expect(match).toMatchSnapshot();
    });

    test('should not be able to create match against yourself', async () => {
        expect.assertions(1);
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        return expect(service.newMatch(1, '1', '1', '1', '2')).rejects.toMatchSnapshot();
    });

    test('should not be able to create match with only a winner and no loser', async () => {
        expect.assertions(1);
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        return expect(service.newMatch(1, '1', '2', '1', null)).rejects.toMatchSnapshot();
    });

    test('should not be able to create match with only a loser and no winner', async () => {
        expect.assertions(1);
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        return expect(service.newMatch(1, '1', '2', null, '2')).rejects.toMatchSnapshot();
    });

    test('should not be able to create match if winner and loser are the same user', async () => {
        expect.assertions(1);
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        return expect(service.newMatch(1, '1', '2', '1', '1')).rejects.toMatchSnapshot();
    });

    test('should not be able to create match if winner and loser are the same user', async () => {
        expect.assertions(1);
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        return expect(service.newMatch(1, '1', '2', '1', '1')).rejects.toMatchSnapshot();
    });

    test("should not be able to create match if users don't match the winner or loser", async () => {
        expect.assertions(1);
        const service = createMatchService(mockMatchRepo, mockLadderUserRepo);
        return expect(service.newMatch(1, '1', '2', '3', '4')).rejects.toMatchSnapshot();
    });
});
