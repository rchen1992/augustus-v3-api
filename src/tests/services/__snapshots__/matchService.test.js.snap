// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Service should be able to get create match 1`] = `
Object {
  "ladder_id": 1,
  "loser_id": "2",
  "match_id": 1,
  "tied": false,
  "user_1_id": "1",
  "user_2_id": "2",
  "winner_id": "1",
}
`;

exports[`Match Service should be able to get create match with a tie 1`] = `
Object {
  "ladder_id": 1,
  "loser_id": null,
  "match_id": 1,
  "tied": true,
  "user_1_id": "1",
  "user_2_id": "2",
  "winner_id": null,
}
`;

exports[`Match Service should be able to get user match stats 1`] = `
Object {
  "lossCount": 0,
  "matchCount": 1,
  "tieCount": 0,
  "winCount": 1,
}
`;

exports[`Match Service should not be able to create match against yourself 1`] = `[Error: Cannot log match against yourself.]`;

exports[`Match Service should not be able to create match if users don't match the winner or loser 1`] = `[Error: Provided users must match either the winner or loser]`;

exports[`Match Service should not be able to create match if winner and loser are the same user 1`] = `[Error: Winner and loser cannot be the same user.]`;

exports[`Match Service should not be able to create match if winner and loser are the same user 2`] = `[Error: Winner and loser cannot be the same user.]`;

exports[`Match Service should not be able to create match with only a loser and no winner 1`] = `[Error: Only one of winner or loser were specified.]`;

exports[`Match Service should not be able to create match with only a winner and no loser 1`] = `[Error: Only one of winner or loser were specified.]`;
